block_darwin.go

```diff
@@ -0,0 +1,250 @@
+	DeviceTreePath                              string // IODeviceTree:/PCI0@0/RP17@1B/ANS2@0/AppleANS2Controller
+	DiskUUID                                    string // some-uuid-foo-bar
+	Ejectable                                   bool   // false
+	EjectableMediaAutomaticUnderSoftwareControl bool   // false
+	EjectableOnly                               bool   // false
+	FilesystemName                              string // APFS
+	FilesystemType                              string // apfs
+	FilesystemUserVisibleName                   string // APFS
+	FreeSpace                                   int64  // 343975677952
+	GlobalPermissionsEnabled                    bool   // true
+	IOKitSize                                   int64  // 499963174912
+	IORegistryEntryName                         string // Macintosh HD
+	Internal                                    bool   // true
+	MediaName                                   string //
+	MediaType                                   string // Generic
+	MountPoint                                  string // /
+	ParentWholeDisk                             string // disk1
+	PartitionMapPartition                       bool   // false
+	RAIDMaster                                  bool   // false
+	RAIDSlice                                   bool   // false
+	RecoveryDeviceIdentifier                    string // disk1s3
+	Removable                                   bool   // false
+	RemovableMedia                              bool   // false
+	RemovableMediaOrExternalDevice              bool   // false
+	SMARTStatus                                 string // Verified
+	Size                                        int64  // 499963174912
+	SolidState                                  bool   // true
+	SupportsGlobalPermissionsDisable            bool   // true
+	SystemImage                                 bool   // false
+	TotalSize                                   int64  // 499963174912
+	VolumeAllocationBlockSize                   int64  // 4096
+	VolumeName                                  string // Macintosh HD
+	VolumeSize                                  int64  // 499963174912
+	VolumeUUID                                  string // some-uuid-foo-bar
+	WholeDisk                                   bool   // false
+	Writable                                    bool   // true
+	WritableMedia                               bool   // true
+	WritableVolume                              bool   // true
+	// also has a SMARTDeviceSpecificKeysMayVaryNotGuaranteed dict with various info
+	// NOTE: VolumeUUID sometimes == DiskUUID, but not always. So far Content is always a different UUID.
+}
+
+type ioregPlist struct {
+	// there's a lot more than just this...
+	ModelNumber  string `plist:"Model Number"`
+	SerialNumber string `plist:"Serial Number"`
+	VendorName   string `plist:"Vendor Name"`
+}
+
+func (ctx *context) getDiskUtilListPlist() (diskUtilListPlist, error) {
```

Comment:
For all of these methods, please change the return value to (*diskUtilListPlist, error). Instead of returning a "blank" `diskUtilListPlist{}` like on lines 100 and 105, instead return `nil, errors.Wrap(err, ...)`

Reply:
Done.